File: packages/android/app/src/main/java/studio/eugenezakharov/opencode/api/models/TodoItem.kt
package studio.eugenezakharov.opencode.api.models

import kotlinx.serialization.Serializable

/**
 * One task in a session's todo list (`GET /session/:id/todo`, `todo.updated`).
 * `status` is pending / in_progress / completed / cancelled. Mirrors iOS `TodoItem`.
 */
@Serializable
data class TodoItem(
    val content: String,
    val status: String,
    val priority: String? = null,
) {
    val done: Boolean get() = status == "completed" || status == "cancelled"
}

File: packages/ios/OpenCode/API/Models/TodoItem.swift
import Foundation

/// One task in a session's todo list (`GET /session/:id/todo`, `todo.updated`).
/// `status` is pending / in_progress / completed / cancelled.
struct TodoItem: Decodable, Identifiable {
    let content: String
    let status: String
    let priority: String?

    var id: String { content }
    var done: Bool { status == "completed" || status == "cancelled" }
}
